.travis.yml

```diff
@@ -68,207 +63,48 @@ addons:
-
-    - stack --no-terminal test --fast --haddock --no-haddock-deps --coverage cardano-wallet-jormungandr:unit
-    - mkdir -p .coverage/jormungandr && find . -name "*.tix" ! -path "*.coverage*" -exec cp \{} .coverage/jormungandr \;
-
-    - stack --no-terminal test --fast --haddock --no-haddock-deps --coverage cardano-wallet-http-bridge:unit
-    - mkdir -p .coverage/http-bridge && find . -name "*.tix" ! -path "*.coverage*" -exec cp \{} .coverage/http-bridge \;
-
-    - tar czf $STACK_WORK_CACHE .stack-work .coverage lib/**/.stack-work
-
-  - stage: integration 🖇️
-    if: branch IN (bors/staging, bors/trying)
-    name: "Integration"
-    script:
-    - travis_retry curl -L https://github.com/KtorZ/cardano-http-bridge/releases/download/v0.0.5/hermes-testnet.tar.gz | tar xz -C $HOME
-
-    - stack --no-terminal test --fast --haddock --no-haddock-deps --coverage cardano-wallet-http-bridge:integration
-    - mkdir -p .coverage/http-bridge && find . -name "*.tix" ! -path "*.coverage*" -exec cp \{} .coverage/http-bridge \;
-
-    - stack --no-terminal test --fast --haddock --no-haddock-deps --coverage cardano-wallet-jormungandr:integration
-    - mkdir -p .coverage/jormungandr && find . -name "*.tix" ! -path "*.coverage*" -exec cp \{} .coverage/jormungandr \;
-
-    - tar czf $STACK_WORK_CACHE .stack-work .coverage lib/**/.stack-work lib/**/*.tix
-
   - stage: deploy 🚀
-    if: branch IN (bors/staging, bors/trying)
-    name: "Coveralls"
-    script:
-    - export LTS=$(cat stack.yaml | grep resolver) # Extract the LTS from the stack.yaml
-    - git clone https://github.com/rubik/stack-hpc-coveralls && cd stack-hpc-coveralls && git checkout 3d8352d5642ab214a7a574bd797880ae39595a44 && echo $LTS > stack.yaml
-    - stack --no-terminal install && cd -
-    - stack --no-terminal install hpc
-    # Ignore modules that are full of Template Haskell auto-generated code
-    - cd lib/core
-    - find $LOCAL_MIX_DIR -type f -name "Cardano.Wallet.DB.Sqlite.TH.mix" > overlay.hpc
-    - 'sed -i "s/.*hpc\/\(.*\).mix/module \"\1\" {}/" overlay.hpc'
-    - stack exec -- hpc overlay --hpcdir $LOCAL_MIX_DIR overlay.hpc > Cardano.Wallet.DB.Sqlite.TH.tix
-    - 'sed -i "s/0,/1,/g" Cardano.Wallet.DB.Sqlite.TH.tix'
-    - cd -
-    - mv lib/core/*.tix .coverage/core
-    # Re-build the coverage report taking .tix from executables running outside of the test suites
-    - stack hpc report .coverage/**/*.tix
-    - shc combined custom || true
-
-  - stage: deploy 🚀
-    if: type != pull_request AND branch = master
+    if: type != pull_request AND (branch = master OR tag =~ ^v)
     name: "Haddock"
     script:
     - mkdir -p haddock/edge api/edge
     - cp -Rv specifications/api/* api/edge
     - mv $(stack path --local-doc-root)/* haddock/edge
     - git checkout --orphan gh-pages-deploy && git reset
     - git add api haddock && git commit -m $TRAVIS_COMMIT
+    - git fetch origin
```

Comment:
Just curious: why do this now, but not previously?

Reply:
Sometimes, deploying the gh-pages branch fails for an unknown reason (we have to run git push in silent mode to avoid leaking the github token on errors...). Yet, I noticed that this often occurred when two builds of the master branch where running concurrently. So my guess is that it can't push to the gh-pages because the remote has changed in between the first fetch and the moment it tries to push a commit on top (because of the other build). So, I am hoping that this harmless `git fetch` would solve this. But that's really a guess.